### PR TITLE
Better checks

### PR DESCRIPTION
**Pull request by tpazderka**
- [ ] Any changes relevant to users are recorded in the `CHANGELOG.md`.
- [ ] The documentation has been updated, if necessary.
---
The CI fails on one error in MyPy which should be fixed by #648.

### CODE REVIEW

File: src/oic/oic/provider.py [hunk 1]
```diff
@@ -258,15 +262,15 @@ def __init__(
...
+        self.keyjar = self.server.keyjar  # type: KeyJar
 
         self.endp.extend([UserinfoEndpoint, RegistrationEndpoint, EndSessionEndpoint])
 
         self.userinfo = userinfo
         self.template_renderer = template_renderer
         self.baseurl = baseurl or name
-        self.cert = []
-        self.cert_encryption = []
+        self.cert = []  # type: ignore  # UNUSED?
```
Comment (schlenk) [inline src/oic/oic/provider.py]: Probably unused, doesn't ring a bell here either. If those are not used in the examples, they could go.

File: src/oic/utils/http_util.py [hunk 2]
```diff
@@ -225,12 +229,16 @@ def extract(environ, empty=False, err=False):
     :param empty: Stops on empty fields (default: Fault)
     :param err: Stops on errors in fields (default: Fault)
     """
+    # FIXME: This seems to be unused -> Deprecate?
```
Comment (schlenk) [inline src/oic/utils/http_util.py]: Thats basic wsgi parsing, so yes, deprecate if not in use.

File: src/oic/utils/keyio.py [hunk 3]
```diff
@@ -451,8 +457,8 @@ def __init__(
             ``requests`` documentation.
         :return:
         """
-        self.spec2key = {}
-        self.issuer_keys = {}
+        self.spec2key = {}  # type: ignore  # FIXME: This seems unused -> Deprecate?
```
Comment (schlenk) [inline src/oic/utils/keyio.py]: yes, deprecate seems ok

Comment (tpazderka) [reply src/oic/utils/keyio.py]: Couldn't find a use so I removed it completely.

File: src/oic/oic/__init__.py [hunk 4]
```diff
@@ -1447,8 +1450,8 @@ def register(self, url, registration_token=None, **kwargs):
 
         headers = {"content-type": "application/json"}
         if registration_token is not None:
-            headers["Authorization"] = b"Bearer " + b64encode(
-                registration_token.encode()
+            headers["Authorization"] = (
+                "Bearer " + b64encode(registration_token.encode()).decode()
```
[the remark this answers is not in the document]
Comment (tpazderka) [reply src/oic/oic/__init__.py]: Wondering myself as well :)

File: src/oic/oic/provider.py (hunk 1 above)
Comment (tpazderka) [reply src/oic/oic/provider.py]: OK, it turns out it is used in example1:
https://github.com/OpenIDC/pyoidc/blob/699cc99850d26565d3fbdc6357ca85a56bd6e5e9/oidc_example/op1/claims_provider.py#L186-L188

But it looks like it shouldn't be used in this way and rather configured statically? It is not set anywhere in the example...

File: src/oic/utils/http_util.py (hunk 2 above)
Comment (tpazderka) [reply src/oic/utils/http_util.py]: Done.

File: src/oic/oic/provider.py (hunk 1 above)
Comment (schlenk) [reply src/oic/oic/provider.py]: Yes, that should just be some static resource.

**Comment by schlenk** [comment]
Looks good, feel free to merge.

**Pull request merged**